src/System.Management.Automation/CoreCLR/CorePsAssemblyLoadContext.cs

```diff
@@ -587,4 +587,27 @@ public static void SetPowerShellAssemblyLoadContext([MarshalAs(UnmanagedType.LPW
             PowerShellAssemblyLoadContext.InitializeSingleton(basePaths);
         }
     }
+
+    /// <summary>
+    /// Provides helper functions to faciliate calling managed code from a native PowerShell host.
+    /// </summary>
+    public static unsafe class PowerShellUnsafeAssemblyLoad
+    {
+        /// <summary>
+        /// Load an assembly in memory from unmanaged code.
+        /// </summary>
+        /// <param name="data">
+        /// Unmanaged pointer to assembly data buffer.
+        /// </param>
+        /// <param name="size">
+        /// Size in bytes of the assembly data buffer.
+        /// </param>
+        [UnmanagedCallersOnly]
+        public static void LoadAssemblyFromMemory(IntPtr data, int size)
+        {
+            Byte* bytes = (Byte*)data.ToPointer();
```

Comment:
Will the follow works?
```suggestion
            Byte* bytes = (Byte*)data;
```

Reply:
as a matter of fact, it turns out to not be necessary. we're now down to a function that has 2 lines of code in it.